### ISSUE TITLE
residentKey key is not supported

### ISSUE THREAD

**Issue by DarkGhostHunter** [web-auth/webauthn-framework#134]
**Describe the bug**
When creating a new instance of `AuthenticatorSelectionCriteria`, the [`residentKey` parameter](https://w3c.github.io/webauthn/#dom-authenticatorselectioncriteria-residentkey) is not supported with any option.

**To Reproduce**
```php

use Webauthn\AuthenticatorSelectionCriteria;

new AuthenticatorSelectionCriteria('cross-platform, 'preferred', 'preferred')
```` 

**Expected behavior**

Return the value of the `residentKey` as part of the attestation JSON:

```json
{
   "authenticatorSelection": {
       "authenticatorAttachment": "cross-platform",
        "residentKey": "preferred",
        "userVerification": "preferred"
    },
    // ...
}
```

**Screenshots**
None.

**Desktop (please complete the following information):**
 - OS: Not applicable
 - Browser Not applicable
 - Version Not applicable

**Smartphone (please complete the following information):**
 - Device: Not applicable
 - OS: Not applicable
 - Browser Not applicable
 - Version Not applicable

**Additional context**
Seems like this is a very simple fix while allowing WebAuthn v1 retro-compatibility. just make a fourth argument.

```php
    public function __construct(?string $authenticatorAttachment = null, 
                                bool $requireResidentKey = false, 
                                string $userVerification = self::USER_VERIFICATION_REQUIREMENT_PREFERRED,
                                string $residentKey = null
    )
```

If the fourth argument is `null`, don't push it inside the JSON.

**Comment by Spomky** [comment]
Hi @DarkGhostHunter,

None of the specification level 2 changes are supported. This is something that is planned.
I keep this issue opened for the moment. I concur with you the modification looks quitte easy to implement.
A dedicated branch will be created for that purpose.

**Comment by Spomky** [comment]
Done.
The changes will be available in `v3.3.0`
